public/js/components/App.js

```diff
@@ -60,8 +60,8 @@ const App = React.createClass({
   },
 
   componentDidMount() {
-    this.shortcuts = new KeyShortcuts({ window });
-    this.shortcuts.on("CmdOrCtrl+P", this.toggleSourcesSearch);
+    const shortcuts = setupShortcuts(window);
```

Comment:
OK, I'll keep looking at this PR a bit too. I bet we could get it merged tomorrow.

Passing a function helps because you're not passing the prop down. The problem before is you're passing `this.shortcuts` down as a value, so if it's `null`, you just pass `null` and child components will forever get `null`. If you pass a function, child components will always get the latest `this.shortcuts` which will not be null after mounting.

But it sound like context might be a bit more promising anyway.